libs/designer/src/lib/core/state/unitTest/unitTestSlice.ts

```diff
@@ -134,6 +134,25 @@ const checkAssertionsErrors = (state: UnitTestState) => {
   }
 };
 
+/**
+ * Revalidates all assertions in the unit test state.
+ * This function clears existing validation errors and performs a fresh validation on all assertions.
+ * @param {UnitTestState} state - The current unit test state.
+ */
+const revalidateAllAssertions = (state: UnitTestState) => {
+  state.validationErrors.assertions = {};
+  Object.values(state.assertions).forEach((assertion) => {
+    const { name, id, assertionString } = assertion;
+    const validationErrors = {
```

Comment:
nit: unless I'm missing something this would have all validation results not just the errors? If so I would change this to validationResults instead of errors.